crates/wamr-sys/build.rs

```diff
@@ -14,12 +14,35 @@ fn main() {
     let wamr_root = wamr_root.join("wasm-micro-runtime");
     assert!(wamr_root.exists());
 
-    let is_espidf = env::var("CARGO_CFG_TARGET_OS").unwrap() != "espidf";
+    let is_espidf = env::var("CARGO_FEATURE_ESP_IDF").is_ok()
+        && env::var("CARGO_CFG_TARGET_OS").unwrap() == "espidf";
 
-    if is_espidf {
+    println!("cargo:rerun-if-env-changed=WAMR_BUILD_PLATFORM");
+    println!("cargo:rerun-if-env-changed=WAMR_SHARED_PLATFORM_CONFIG");
+    println!("cargo:rerun-if-env-changed=CARGO_FEATURE_ESP_IDF");
+
+    if is_espidf
+        && (env::var("WAMR_BUILD_PLATFORM").is_ok()
+            || env::var("WAMR_SHARED_PLATFORM_CONFIG").is_ok())
+    {
+        panic!("ESP-IDF build cannot use custom platform build (WAMR_BUILD_PLATFORM) or shared platform config (WAMR_SHARED_PLATFORM_CONFIG)");
+    }
+
+    if !is_espidf {
         let enable_llvm_jit = if cfg!(feature = "llvmjit") { "1" } else { "0" };
         // TODO: define LLVM_DIR
-        let dst = Config::new(&wamr_root)
+        let mut dst = Config::new(&wamr_root);
+
+        if let Ok(platform_name) = env::var("WAMR_BUILD_PLATFORM") {
+            dst.define("WAMR_BUILD_PLATFORM", &platform_name);
+        }
+
+        if let Ok(platform_config) = env::var("WAMR_SHARED_PLATFORM_CONFIG") {
+            dst.define("SHARED_PLATFORM_CONFIG", &platform_config);
+            println!("cargo:rerun-if-changed={}", platform_config);
+        }
+
```

Comment:
This is not for esp-idf. right? It is for other new-os.

Reply:
like macos, windows, android, those are OSes can be represented by `CARGO_CFG_TARGET_OS`(https://doc.rust-lang.org/reference/conditional-compilation.html#target_os), and those platform configuration files (shared_platform.cmake) are listed in WAMR repo(core/shared/platform/XXX/. No need to create). 

Therefore, I am thinking that maybe we can set both WAMR_BUILD_PLATFORM and WAMR_SHARE_PLATFORM_CONIG for those OSes. 

For example, create a pre-defined dict which keys are possible values(and WAMR supported) of target_os and values are paths for correspond *shared_platform.cmake*. And firstly use user defined env variables(allow overwrite). If not set, use pre-defined dict.

After above, I guess users can use wamr-rust-sdk in other OSes.